File: src/main/java/de/danielmaile/aether/item/ListenerConverter.java
package de.danielmaile.aether.item;

import de.danielmaile.aether.Aether;
import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryOpenEvent;
import org.bukkit.inventory.ItemStack;

public class ListenerConverter implements Listener
{
    @EventHandler
    public void onInventoryOpen(InventoryOpenEvent event)
    {
        if (!Aether.getConfigManager().isItemConverterEnabled()) return;
        ItemStack[] content = event.getInventory().getContents();
        for (int i = 0; i < content.length; i++)
        {
            ItemStack itemStack = content[i];
            if (itemStack == null || itemStack.getType() == Material.AIR) continue;

            ItemType type = ItemType.getFromTag(itemStack);
            if (type == null) continue;

            ItemStack typeStack = type.getItemStack(itemStack.getAmount());
            if (itemStack.equals(typeStack)) continue;

            event.getInventory().setItem(i, typeStack);
        }
    }
}

File: src/main/java/de/danielmaile/aether/item/ListenerConverter.kt
package de.danielmaile.aether.item

import de.danielmaile.aether.inst
import de.danielmaile.aether.item.ItemType.Companion.fromTag
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryOpenEvent
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.inventory.Inventory

class ListenerConverter : Listener {

    @EventHandler
    fun onInventoryOpen(event: InventoryOpenEvent) {
        if (inst().configManager.itemConverterEnabled) {
            checkAndReplace(event.inventory)
        }
    }

    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        if (inst().configManager.itemConverterEnabled) {
            checkAndReplace(event.player.inventory)
            checkAndReplace(event.player.enderChest)
        }
    }

    private fun checkAndReplace(inventory: Inventory) {
        val content = inventory.contents ?: return

        for (i in content.indices) {
            val itemStack = content[i] ?: continue
            val type = fromTag(itemStack) ?: continue
            val typeStack = type.getItemStack(itemStack.amount)

            if (itemStack == typeStack) continue
            inventory.setItem(i, typeStack)
        }
    }
}